### PR TITLE
`Programming exercises`: Adjust translations for participation mode tooltips

### PR DESCRIPTION
**Pull request by iyannsch**
> [!TIP]
> Most basic functionality is already merged so this PR only includes the leftover review comments :)

#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [ ] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

#### Changes affecting Programming Exercises
- [x] **High priority**: I tested **all** changes and their related features with **all** corresponding user types on a test server configured with the **integrated lifecycle setup** (LocalVC and LocalCI).
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server configured with **Gitlab** and **Jenkins**.

### Motivation and Context

As the amount of participation modes grow from `Offline IDE` and `Online Editor` to now also include the `Online IDE`, @bassner mentioned that he'd love a description for those options. Most basic functionality is already merged so this PR only includes the leftover review comments :)

### Description

This PR adds tooltips to the three participations modes.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to a course
3. Create or edit a programming exercise
4. Verify that the participations modes have tooltips

#### Exam Mode Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to a course
3. Create or edit a programming exercise
4. Verify that the participations modes have tooltips

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review

- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance even for very large courses with more than 2000 students.
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance even for very large courses with more than 2000 students.
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2
#### Exam Mode Test
- [x] Test 1
- [ ] Test 2
#### Performance Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

Screenshot 13 10 - 15 25

## Summary by CodeRabbit

- **New Features**
	- Introduced contextual help icons for "allowOfflineIde" and "allowOnlineEditor" checkboxes to enhance user experience.
	- Added detailed descriptions for IDE options in both English and German to improve clarity for users.

These updates aim to provide better guidance and usability for students interacting with programming exercises.

### CODE REVIEW

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 1]
```diff
@@ -118,18 +118,21 @@
...
             "allowOnlineEditor": {
                 "title": "Online-Editor erlauben",
+                "description": "Ein Online-Texteditor mit Datei-Explorer und Code-Hervorhebungsfunktionen, jedoch ohne die Möglichkeit, Code zu kompilieren oder auszuführen.",
                 "alert": "Es muss mindestens eine Option (Offline-IDE, Online-Editor oder Online-IDE) ausgewählt sein",
                 "alertNoTheia": "Es muss mindestens eine Option (Offline-IDE oder Online-Editor) ausgewählt sein"
             },
             "onlineEditor": "Online",
             "allowOfflineIde": {
                 "title": "Offline-IDE erlauben",
+                "description": "Aktivieren Sie diese Option, um es zu erlauben, die Übung herunterzuladen und lokal mit der bevorzugten IDE daran zu arbeiten.",
```
Comment (pzdr7) [inline src/main/webapp/i18n/de/programmingExercise.json:128]: We usually use the informal "Du" over "Sie"

https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines#german-language

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 2]
```diff
@@ -118,18 +118,21 @@
...
                 "alert": "Es muss mindestens eine Option (Offline-IDE, Online-Editor oder Online-IDE) ausgewählt sein",
                 "alertNoTheia": "Es muss mindestens eine Option (Offline-IDE oder Online-Editor) ausgewählt sein"
             },
             "offlineIde": "IDE",
             "allowOnlineIde": {
                 "title": "Online-IDE erlauben",
+                "description": "Eine auf VisualStudioCode basierende Online-Entwicklungsumgebung mit Code-Hervorhebungs- und Formatierungsfunktionen. Vorkonfiguriert für die Sprache der Übung und mit der Möglichkeit, Code online zu kompilieren oder auszuführen.",
```
Comment (pzdr7) [inline src/main/webapp/i18n/de/programmingExercise.json:135]: I think Microsoft stylizes it as "Visual Studio Code" https://code.visualstudio.com/

```suggestion
                "description": "Eine auf Visual Studio Code basierende Online-Entwicklungsumgebung mit Code-Hervorhebungs- und Formatierungsfunktionen. Vorkonfiguriert für die Sprache der Übung und mit der Möglichkeit, Code online zu kompilieren oder auszuführen.",
```

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 3]
```diff
@@ -130,18 +130,21 @@
...
             "allowOfflineIde": {
                 "title": "Allow Offline IDE",
+                "description": "Check this option to allow students to download the exercise and work on it locally using their preferred IDE.",
                 "alert": "At least one option (Offline IDE, Online Editor, or Online IDE) must be selected",
                 "alertNoTheia": "At least one option (Offline IDE or Online Editor) must be selected"
             },
             "offlineIde": "IDE",
             "allowOnlineIde": {
                 "title": "Allow Online IDE",
+                "description": "An online coding environment based on VisualStudioCode featuring code highlighting and formatting features. Pre-configured to your exercise's language and with the ability to compile or run code online.",
```
Comment (pzdr7) [inline src/main/webapp/i18n/en/programmingExercise.json:147]:
```suggestion
                "description": "An online coding environment based on Visual Studio Code featuring code highlighting and formatting features. Pre-configured to your exercise's language and with the ability to compile or run code online.",
```

**Review by farisd16 — DISMISSED** [review]
tested on ts3, I could see two participation modes and tooltips next to them

**Review by cremertim — DISMISSED** [review]
lgtm

**Review by HawKhiem — DISMISSED** [review]
Works great

**Review by pzdr7 — DISMISSED** [review]
Code 👍

**Review by asliayk — DISMISSED** [review]
code lgtm

**Review by magaupp — DISMISSED** [review]
Tested on TS9 (Theia enabled). All 3 tooltips are shown while editing an exercise.

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 4]
```diff
@@ -118,18 +118,21 @@
...
             "allowOnlineEditor": {
                 "title": "Online-Editor erlauben",
+                "description": "Ein Online-Texteditor mit Datei-Explorer und Code-Hervorhebungsfunktionen, jedoch ohne die Möglichkeit, Code zu kompilieren oder auszuführen.",
                 "alert": "Es muss mindestens eine Option (Offline-IDE, Online-Editor oder Online-IDE) ausgewählt sein",
                 "alertNoTheia": "Es muss mindestens eine Option (Offline-IDE oder Online-Editor) ausgewählt sein"
             },
             "onlineEditor": "Online",
             "allowOfflineIde": {
                 "title": "Offline-IDE erlauben",
+                "description": "Aktiviere diese Option, um es zu erlauben, die Übung herunterzuladen und lokal mit der bevorzugten IDE daran zu arbeiten.",
```
Comment (magkue) [inline src/main/webapp/i18n/de/programmingExercise.json:128]:
```suggestion
                "description": "Erlaubt Studenten die Übung mit Git herunterzuladen und diese lokal mit der bevorzugten IDE zu bearbeiten.",
```

File: src/main/webapp/i18n/en/programmingExercise.json [hunk 5]
```diff
@@ -130,18 +130,21 @@
...
             "allowOnlineEditor": {
                 "title": "Allow Online Editor",
+                "description": "An online text editor featuring a file explorer, coding window, and code highlighting features but without the ability to compile or run code.",
                 "alert": "At least one option (Offline IDE, Online Editor, or Online IDE) must be selected",
                 "alertNoTheia": "At least one option (Offline IDE or Online Editor) must be selected"
             },
             "onlineEditor": "Online",
             "allowOfflineIde": {
                 "title": "Allow Offline IDE",
+                "description": "Check this option to allow students to download the exercise and work on it locally using their preferred IDE.",
```
Comment (magkue) [inline src/main/webapp/i18n/en/programmingExercise.json:140]:
```suggestion
                "description": "Enables students to download the exercise via Git and to work on it locally using their preferred IDE.",
```

**Review by magkue — DISMISSED** [review]
Code lgtm. Small (and optional) suggestion to improve clarity.

File: src/main/webapp/i18n/de/programmingExercise.json [hunk 6]
```diff
@@ -118,18 +118,21 @@
...
                 "title": "Online-Editor erlauben",
+                "description": "Ein Online-Texteditor mit Datei-Explorer und Code-Hervorhebungsfunktionen, jedoch ohne die Möglichkeit, Code zu kompilieren oder auszuführen.",
                 "alert": "Es muss mindestens eine Option (Offline-IDE, Online-Editor oder Online-IDE) ausgewählt sein",
                 "alertNoTheia": "Es muss mindestens eine Option (Offline-IDE oder Online-Editor) ausgewählt sein"
             },
             "onlineEditor": "Online",
             "allowOfflineIde": {
                 "title": "Offline-IDE erlauben",
+                "description": "Erlaubt Studenten die Übung mit Git herunterzuladen und diese lokal mit der bevorzugten IDE zu bearbeiten.",
                 "alert": "Es muss mindestens eine Option (Offline-IDE, Online-Editor oder Online-IDE) ausgewählt sein",
```
[the remark this answers is not in the document]
Comment (magkue) [reply src/main/webapp/i18n/de/programmingExercise.json:129]: @iyannsch Oh my bad, that should definitely be adjusted

**Review by magaupp — DISMISSED** [review]
Tested on TS9: Still working :+1:

**Review by edkaya — DISMISSED** [review]
tested on ts9, works great 👍

**Review by HawKhiem — DISMISSED** [review]
tested on server 6. Still works perfectly

**Review by sachmii — APPROVED** [review]
Tested on TS4 also in exam mode. Tooltips show up, works fine.

**Review by az108 — APPROVED** [review]
Code 👍

**Review by pzdr7 — DISMISSED** [review]
Reapprove

**Review by magaupp — DISMISSED** [review]
Tested on TS9. Still works.

Looks like most changes made it into #9471.

**Review by krusche — APPROVED** [review]

**Pull request merged**